pennylane/pulse/transmon.py

```diff
@@ -0,0 +1,206 @@
+
+    where :math:`[a^\dagger_p, a_q] = i \delta_{pq}` are bosonic creation and annihilation operators.
+    The first term describes the dressed qubit frequencies :math:`\omega_q`, the second term their
+    coupling :math:`g_{ij}` and the last the anharmonicity :math:`\alpha_q`, which all can vary for
+    different qubits. In practice, the bosonic operators are restricted to a finite dimension of the
+    local Hilbert space (default ``d=2`` corresponds to qubits).
+    In that case, the anharmonicity is set to :math:`\alpha=0` and ignored.
+
+    The values of :math:`\omega \pm \alpha` are typically around :math:`(5 \pm 0.3) \times 2\pi \text{GHz}` and it is common
+    for different qubits to be out of tune with different energy gaps. The coupling strength
+    :math:`g` typically varies betwewen :math:`[0.001, 0.1] \times 2\pi \text{GHz}`. For some example parameters,
+    see e.g. `arXiv:1804.04073 <https://arxiv.org/abs/1804.04073>`_,
+    `arXiv:2203.06818 <https://arxiv.org/abs/2203.06818>`_, or `arXiv:2210.15812 <https://arxiv.org/abs/2210.15812>`_.
+
+    .. note:: Currently only supporting ``d=2`` with qudit support planned in the future.
+
+    .. seealso::
+
+        :func:`~.drive`
+
+    Args:
+        omega (Union[float, list[float]]): List of dressed qubit frequencies in GHz.
+            When passing a single float all qubits are assumed to have that same frequency.
+        connections (list[tuple(int)]): List of connections ``(i, j)`` between qubits i and j.
+        g (Union[float, list[float]]): List of coupling strengths in GHz. Needs to match the length of ``connections``.
+            When passing a single float need explicit ``wires``.
+        anharmonicity (Union[float, list[float]]): List of anharmonicities in GHz. Ignored when ``d=2``.
+            When passing a single float all qubits are assumed to have that same anharmonicity.
+        wires (list): Optional, defaults to ``range(len(omega))`` when set to ``None``. When passing explicit ``wires``,
+            needs to be of the same length as omega. Note that there can be additional wires in the resulting operator from
+            the ``connections``, which are treated independently.
+        d (int): Local Hilbert space dimension. Defaults to ``d=2`` and is currently the only supported value.
+
+    Returns:
+        HardwareHamiltonian: a :class:`~.ParametrizedHamiltonian` representing the transmon interaction
+
+    **Example**
+
+    We can set up the transmon interaction Hamiltonian with uniform coefficients by passing ``float`` values.
+
+    .. code-block::
+
+        connections = [[0, 1], [1, 3], [2, 1], [4, 5]]
+        H = qml.pulse.transmon_interaction(omega=0.5, connections=connections, g=1.)
+
+    The resulting :class:`~.ParametrizedHamiltonian` consists of ``4`` coupling terms and ``6`` qubits
+    because there are six different wire indices in ``connections``.
+
+    >>> print(H)
+    ParametrizedHamiltonian: terms=10
```

Comment:
I suppose this is the motivation for wanting a better repr for `ParametrizedHamiltonian`?

Reply:
One of them yes. I also just often noticed when prototyping things that it would be handy to see the terms immediately (without having to call it and then further be stuck with the unreadable `(((((((3*(1j*(qml.PauliX(0))+...` notation